src/JustEat.StatsD/Transport/Data.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+
+namespace JustEat.StatsD
+{
+    public static class DataExtensions
+    {
+        public static ReadOnlySpan<byte> GetSpan(this in Data src)
+        {
+            if (src.Array == null)
+            {
+                return src.Span;
+            }
+            return src.Array.AsSpan();
+        }
+
+        public static byte[] GetArray(this in Data src)
+        {
+            if (src.Array == null)
+            {
+                return src.Span.ToArray();
+            }
+            return src.Array;
+        }
+    }
+
+    public ref struct Data
```

Comment:
So is this basically being a C-style union?

Reply:
It is, the idea is to allow both runtimes to get a unifying type. The major quirk is that Span.ToArray() will allocate on the heap, so should be avoided when Data was created from Span.